tests/ui/thread-local/thread-local-issue-37508.rs

```diff
@@ -28,6 +28,7 @@ pub unsafe fn set() -> Option<&'static BB> {
         return None;
     }
     Some(&KEY.inner)
+    //~^ WARN creating a shared reference to mutable static is discouraged [static_mut_refs]
```

Comment:
Since this test is unrelated, let's fix (or `allow(..)` and mark `FIXME`) this and other similar uses instead.